fairscale/nn/misc/flatten_params_wrapper.py

```diff
@@ -15,6 +15,37 @@
     from collections import OrderedDict  # noqa: F401
 
 
+class FlatParameter(nn.Parameter):
+    """ A parameter that is initialized from a list of parameters and can be
+        turned into a list of views as needed.
+    """
+
+    def __new__(cls, params: Sequence[nn.Parameter], requires_grad: bool = True) -> "FlatParameter":
+        if not isinstance(params, (list, tuple)) or len(params) == 0:
+            raise ValueError("An non-empty list or tuple argument is needed")
+
+        if not all(isinstance(p, nn.Parameter) for p in params):
+            raise ValueError("List items need to be Parameter types")
+
+        if any(isinstance(p, FlatParameter) for p in params):
+            raise ValueError("Nesting FlatParameter is not supported")
```

Comment:
This guard checks that whether we actually use nested FPW outside of the unit test of FPW itself. CI is running. My local test seems to show that it is not used by FSDP.

Reply:
I assume nested FSDP, where root and children have `flatten_parameters=True`, does not hit this?

Reply:
Right, it doesn't hit. nested FSDP is very common in the unit tests.